client/src/store/plot.js

```diff
@@ -94,21 +94,23 @@ export default {
     PLOT_AVAILABLE_TIME_STEPS_CHANGED({ commit, rootGetters }, timeSteps) {
       commit("PLOT_AVAILABLE_TIME_STEPS_SET", timeSteps);
 
-      let timeStep = rootGetters.VIEW_TIME_STEP;
-      let minTimeStep = rootGetters.VIEW_MIN_TIME_STEP;
-      let maxTimeStep = rootGetters.VIEW_MAX_TIME_STEP;
-      let tsMin = Math.min(...timeSteps);
-      let tsMax = Math.max(...timeSteps);
-
-      let newMin =
-        maxTimeStep < minTimeStep ? tsMin : Math.min(minTimeStep, tsMin);
-      let newMax =
-        maxTimeStep < minTimeStep ? tsMax : Math.max(maxTimeStep, tsMax);
-      let newCurr = Math.max(Math.min(timeStep, newMax), newMin);
+      let newMin = Infinity;
+      let newMax = -Infinity;
+      rootGetters.VIEW_SELECTIONS.forEach((id) => {
+        let steps = rootGetters[`${id}/PLOT_AVAILABLE_TIME_STEPS`] || [];
+        let tsMin = Math.min(...steps);
+        let tsMax = Math.max(...steps);
```

Comment:
Hopefully the size of steps remains manageable as that kind of notation blow in memory with large arrays.

Reply:
It is certainly not guaranteed to remain manageable. I think that this will work for now but I think that long-term this is better as metadata that we can grab for each variable.

Reply:
You probably should create a helper function to return [min, max]